desktop/src/main/java/bisq/desktop/components/paymentmethods/RevolutForm.java

```diff
@@ -125,8 +126,23 @@ public void addFormForDisplayAccount() {
 
     @Override
     public void updateAllInputsValid() {
-        allInputsValid.set(isAccountNameValid()
-                && validator.validate(account.getUserName()).isValid
+        String userName = getUserName();
+        allInputsValid.set(inputValidator.validate(userName).isValid
+                && validator.validate(userName).isValid
                 && account.getTradeCurrencies().size() > 0);
     }
+
+    @Override
+    public PaymentAccount getPaymentAccountForAccountCreation() {
+        String userName = getUserName();
+        if (!userName.isEmpty()) {
+            account.setUserName(userName);
+        }
+        return super.getPaymentAccount();
```

Comment:
Better use account instead of super.getPaymentAccount(); as that would suggest that this is different than the visible account we used for setting the username.